File: swingtest/src/Vector.java
public class Vector {
    public double x, y;
    
    public Vector(double x, double y) {
        this.x = x; this.y = y;
    }
    
    public Vector plus(Vector other) {
        return new Vector(this.x + other.x, this.y + other.y);
    }
    
    public Vector minus(Vector other) {
        return new Vector(this.x - other.x, this.y - other.y);
    }
    
    public double getLength() {
        return Math.sqrt(x * x + y * y);
    }
    
    public Vector times(double factor) {
        return new Vector(this.x * factor, this.y * factor);
    }
    
    public boolean equals(Vector other) {
        return (this.x == other.x && this.y == other.y);
    }
    
    public static double getAngle(Vector start, Vector end) {
        if (start.equals(end)) {
            return 0;
        }
        
        double opp = start.y - end.y;
        double hyp = start.minus(end).getLength();
        double angle = Math.asin(opp / hyp);
        
        if (start.x > end.x) {
            angle = Math.PI - angle;
        }
        
        return angle;
    }
}

File: art_life/swift/Sources/artificial_life/Vector.swift
import Foundation

struct Vector : Equatable {
    var x : Int
    var y : Int
    
    init(_ x : Int, _ y : Int) {
        self.x = x; self.y = y
    }
    
    func plus(other : Vector) -> Vector {
        return Vector(self.x + other.x, self.y + other.y)
    }
    
    func times(factor : Double) -> Vector {
        return Vector(Int(Double(self.x) * factor), Int(Double(self.y) * factor))
    }
    
    static func == (lhs: Vector, rhs: Vector) -> Bool {
        return lhs.x == rhs.x && lhs.y == rhs.y
    }
}